cmdline/diff.py

```diff
@@ -0,0 +1,91 @@
+#!python
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Output a summary table for neuroimaging files (resolution, dimensionality, etc.)
+"""
+from __future__ import division, print_function, absolute_import
+
+import sys
+from optparse import OptionParser, Option
+
+import numpy as np
+
+import nibabel as nib
+import cmdline.utils
+from cmdline.utils import _err, verbose, table2string, ap, safe_get
+import fileinput
+
+__author__ = 'Yaroslav Halchenko & Christopher Cheng'
+__copyright__ = 'Copyright (c) 2017 NiBabel contributors'
+__license__ = 'MIT'
+
+# these fields are processed by the __get method
+header_fields = ['sizeof_hdr', 'dim_info', 'dim', 'intent_p1', 'intent_p2', 'intent_p3', 'intent_code', 'datatype',
+                 'bitpix', 'slice_start', 'pixdim', 'vox_offset', 'scl_slope', 'scl_inter', 'slice_end', 'slice_code',
+                 'xyzt_units', 'cal_max', 'cal_min', 'slice_duration', 'toffset', 'descrip', 'aux_file', 'qform_code',
+                 'sform_code', 'quatern_b', 'quatern_c', 'quatern_d', 'qoffset_x', 'qoffset_y', 'qoffset_z', 'srow_x',
+                 'srow_y', 'srow_z', 'intent_name', 'magic']
```

Comment:
why did you need to explicitly list them?

Reply:
the .header.keys() way of accessing headers you gave to me doesn't work with certain headers - I troubleshot them and there's no problem now that I've individually listed the ones that work. Is that a problem?

Reply:
"doesn't work" is not descriptive - provide summary what exactly "isn't working". If there would be a legit reason to skip, we could just skip some of those - try/except around handling

Hard-listing makes code not generic (might be a good time to revisit beginning of fluent PYTHON book). It wouldn't allow to compare other formats with headers, it other versions of nifti file format.

Reply:
Actually it's working now so forget about the list. Noted on the hard-listing part though, I will keep that in mind for the future! Gotta focus on the bigger picture, got it! Thanks Yarik